### ISSUE TITLE
VS Code ESLint extension: No diagnostics for .md files with @eslint/markdown (Flat Config); CLI works

### ISSUE THREAD

**Issue by CyberT33N** [microsoft/vscode-eslint#2086]
### Summary
When using the VS Code ESLint extension with ESLint v9 flat config and `@eslint/markdown`, Markdown files (`**/*.md`) receive no diagnostics in the editor. Running ESLint via CLI reports the expected `markdown/*` findings (and, when using the processor, code‑block findings). The extension appears not to start linting for `.md` files, even though Markdown support is documented and enabled.

### Context & Background
- ESLint v9 flat config supports a Markdown “language” mode and processors (code‑block extraction).
- `@eslint/markdown` provides:
  - a “language” config for `**/*.md` (e.g., `language: 'markdown/gfm'`) to run `markdown/*` rules on the Markdown AST, and
  - a processor for code‑block linting (e.g., `**/*.md/**`).
- The plugin maintainers confirmed it’s not currently possible to run both flows in a single run (“combined mode”). However, independently of that limitation:
  - The VS Code extension should still display Markdown document diagnostics when only the language config is used, and
  - It should display code‑block diagnostics when the processor is used.
- In the editor, neither appears to happen for `.md` files in the described setup, whereas the CLI does report diagnostics.

A related discussion exists with `@eslint/markdown` maintainers indicating combined mode is not supported; they suggested using default settings or contacting the VS Code ESLint extension maintainers for editor‑side behavior.

### Environment
- Project version: Not determined (consumer project)
- ESLint: 9.36.0
- `@eslint/markdown`: 7.4.0
- VS Code: Not determined
- VS Code ESLint extension: Not determined
- Node:
  - CLI: 22.14.0
  - VS Code ESLint server runtime: 22.9 (observed in the extension’s Output/Debug channel)
- OS/Platform: Linux `6.8.0-83-generic`

### Affected Versions / Branch
- Affects the VS Code ESLint extension with ESLint v9 flat config and `@eslint/markdown` 7.4.0
- Introduced in: Not determined

### Steps to Reproduce
1. Configure ESLint flat config to enable Markdown language rules for `**/*.md`:
   ```ts
   // eslint.config.ts (excerpt)
   import eslintPluginMarkdown from '@eslint/markdown';

   export default [
     {
       files: ['**/*.md', '**/*.mdx'],
       language: 'markdown/gfm',
       plugins: { markdown: eslintPluginMarkdown },
       rules: {
         // any markdown/* rules, e.g. 'markdown/no-html': 'error'
       }
     }
   ];
   ```
2. Ensure the VS Code ESLint extension is enabled and (if applicable) add `"markdown"` to `eslint.probe` or `eslint.validate`. Enable the Debug output channel (`eslint.debug: true`) to observe activity.
3. Open a Markdown file that has clear `markdown/*` violations (e.g., duplicate headings, missing heading space).
4. Observe that in VS Code, the Problems panel shows no diagnostics for the `.md` file, and no inline squiggles appear.
5. Run ESLint via the CLI against the same file (`pnpm eslint README.md`) and observe that diagnostics are correctly reported by the CLI.

Optional variant (code‑block processor):
6. Configure the processor flow to lint code blocks (per `@eslint/markdown` docs) and repeat steps 3–5. CLI shows code‑block diagnostics; editor still shows none in the `.md` file.

### Expected Behavior
- The VS Code ESLint extension should surface the same diagnostics as the CLI:
  - When configured with `language: 'markdown/gfm'` for `**/*.md`, show applicable `markdown/*` diagnostics on the Markdown document.
  - When configured with the processor flow, show diagnostics for fenced code blocks extracted from the Markdown file.

### Actual Behavior
- No diagnostics appear in VS Code for `.md` files despite a working CLI configuration.
- The ESLint Debug output indicates the extension is active and uses Node 22.9, but `.md` buffers receive no linting feedback (no Problems entries, no inline diagnostics).

### Impact & Severity
- Severity: Major
- Developers cannot rely on the editor to enforce Markdown document structure or code‑block quality; teams must fall back to CLI/CI‑only workflows, reducing feedback speed and developer experience.

### Logs & Evidence
```text
- CLI: Reports markdown/* problems (and code‑block problems with processor) for .md files.
- VS Code: No visible diagnostics for the same .md files, even with Debug enabled and extension active.
- Exact VS Code ESLint extension and VS Code versions: Not determined.
```

### Code References
- Markdown flat config enabling `language: 'markdown/gfm'`:
```174:191:src/modules/configs/file-formats/markdown.ts
const upstreamRules = eslintPluginMarkdown.configs.recommended.find(
    (config: ReadonlyDeep<TsEsLintFlatConfigItem>) => isDefined(config.rules)
)?.rules

const upstreamConfig = {
    files: MARKDOWN_FILES_PATTERNS,
    language: 'markdown/gfm',
    languageOptions: {
        frontmatter: 'yaml'
    },
    name: `${CONFIG_NAMESPACE}:${CONFIG_UPSTREAM_NAME}`,
    plugins: { markdown: eslintPluginMarkdown },
    rules: {
        ...upstreamRules
    }
} as const satisfies TsEsLintFlatConfigItem
```

- Markdown file patterns used in the config:
```84:86:src/constants/patterns/internal/constants/files.ts
/** @example `['**/*.md', '**/*.mdx']` */
const MARKDOWN_FILES_PATTERNS = FILE_EXTENSIONS_MARKDOWN.map(ext => `**/*.${ext}`)
```

### Proposed Direction / Next Steps
- Confirm whether the VS Code ESLint extension currently validates `.md` documents by default when a flat config provides `language: 'markdown/...')` for `**/*.md`.
- If not supported, consider:
  - Enabling document validation for Markdown buffers when a workspace flat config includes a Markdown language entry.
  - Documenting the precise settings required (e.g., `eslint.probe`/`eslint.validate` entries for `'markdown'`) and any flat‑config prerequisites (no additional configuration files).
- If supported, identify why the extension does not start linting `.md` buffers in this setup while the CLI does, and provide guidance or a fix.

### Acceptance Criteria
- [ ] Opening a `.md` file in VS Code triggers ESLint validation when a flat config with `language: 'markdown/gfm'` is present.
- [ ] Editor diagnostics match CLI output for `markdown/*` rules.
- [ ] If processor is used, code‑block diagnostics appear in the editor (or documented limitation/workaround is provided).
- [ ] Clear documentation on any required extension settings for Markdown (e.g., `eslint.probe`/`eslint.validate`) with flat config.
- [ ] No breaking changes to existing JavaScript/TypeScript validation.

**Comment by dbaeumer** [comment]
Markdown should actually be auto detected. Can you please provide me with a GitHub repository I can clone with a minimal setup that demos what you are seeing.

**Comment by CyberT33N** [comment]
Sure :) Thank you for helping!

- https://github.com/CyberT33N/markdown-test

**Comment by dbaeumer** [comment]
@CyberT33N 

I cloned the repository and followed your steps and it works for me:

<img width="1958" height="1220" alt="Image" src="https://github.com/user-attachments/assets/1f4c72e9-b2d0-47e8-b044-f3540d1abc25" />

Is there anything special in the ESLint Output channel?

**Comment by CyberT33N** [comment]
Thank you very much for your work and for checking this. I’m really sorry — I HAD TO do it earlier. It must be caused by some installed extensions or my user settings file. When I delete the VSCode folder from my home directory — where the settings are — and completely reset the user settings, it works. I’ll keep trying to figure out what’s causing the complications, in case others run into the same issue later, and I’ll get back to you.

**Comment by CyberT33N** [comment]
The problem is caused by:
- https://marketplace.visualstudio.com/items?itemName=Nuxt.mdc

It can not work together with Eslint Extension for md files

**Comment by dbaeumer** [comment]
@CyberT33N thanks for reporting this back.
